api/app/controllers/spree/api/v2/base_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Spree
+  module Api
+    module V2
+      class BaseController < ActionController::API
+        include CanCan::ControllerAdditions
+
+        private
+
+        def spree_current_store
+          @spree_current_store ||= Spree::Store.current(request.env['SERVER_NAME'])
+        end
+
+        def spree_current_user
+          @spree_current_user ||= Spree.user_class.find_by(id: doorkeeper_token.resource_owner_id) if doorkeeper_token
+        end
+
+        def spree_authorize!(action, subject, *args)
+          authorize!(action, subject, *args)
+        end
+
+        # Needs to be overriden so that we use Spree's Ability rather than anyone else's.
+        def current_ability
+          @current_ability ||= Spree::Ability.new(spree_current_user)
+        end
+
+        def order_token
+          request.headers['X-Spree-Order-Token'] || params[:order_token]
+        end
+
+        def current_order
```

Comment:
@bbonislawski maybe we should prefix this with `spree_` like `current_user`, `current_store` or drop the prefix for those 2? It should be consistent